### DISCUSSION TITLE
Translations

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#116] [General]
merci de m'indiquer les coquilles ..
[Manuel-BiblioteQ-version-1.17-Français.pdf](https://github.com/textbrowser/biblioteq/files/7470175/Manuel-BiblioteQ-version-1.17-Francais.pdf)
.

**Comment by textbrowser** [comment]
Thank you. Added to the repository in Documentation/Contributed.

**Comment by meteos77** [comment]
it is me who thanks you for what you do.
I don't forget that you are the builder of the house, I only do a small part of the decoration
the odt file of the french version compressed
[Manuel-BiblioteQ-version-1.17-Français.odt.tar.gz](https://github.com/textbrowser/biblioteq/files/7472649/Manuel-BiblioteQ-version-1.17-Francais.odt.tar.gz)

**Comment by textbrowser** [comment]
Thank you. Adding now.

**Comment by meteos77** [comment]
the more I use the application the more I have suggestions ...

for the help menu
added local help (machine without network)
added keyboard shortcut F1 to get this help document
and now that the documentation is available in French I would like to have this link that points to the local file in the language of the application on my machine namely /usr/local/bibilioteq/Documentation/BiblioteQ-Fr.pdf
thank you in advance for the integration

**Comment by textbrowser** [comment]
Finished adding a new documentation dialog. It will feature the official document in HTML. As there are English and French versions, both will be accessible. The executable will be larger but that is preferred over complicated packages.

**Reply by meteos77** [reply]
you can indicate that to have access to the documentation offline you must
1 - get the documentation from https://github.com/textbrowser/biblioteq/tree/master/Documentation/Contributed/

2 - choose your preferred language and copy it to the local path

this would avoid having "executable balloon :-)"

**Reply by textbrowser** [reply]
Yup. All available translations and the English version will be distributed and available regardless of the preferred language.

![Screenshot_20211107_083007](https://user-images.githubusercontent.com/10701156/140647130-22ed14f9-1c13-4f4d-b449-4e19d14cbf7c.png)

**Comment by meteos77** [comment]
thank you for the functionality

**Comment by textbrowser** [comment]
English documentation completed.

**Comment by textbrowser** [comment]
And French. :)

![Screenshot_20211105_182048](https://user-images.githubusercontent.com/10701156/140584840-9ee28ef3-9966-4e15-ac5c-4ef11135bc71.png)

**Reply by meteos77** [reply]
very nice work for the translation to be valued

answers to point e) the translation is not fully enhanced.

**Comment by textbrowser** [comment]
The documentation is now loaded from files instead of being embedded into the executable. The executable will balloon.

**Comment by textbrowser** [comment]
Also available on Android.

**Comment by meteos77** [comment]
fichier pour application en Français  c'est pas du 100% , reste la partie  cd , dvd, photographie et des coquilles gramaticale à droite à gauche que je corrigerai au fil de l'eau.

file for application in French
file:///home/yo/Bureau/10-BiblioteQ/2021-11-08_biblioteq_fr_FR-10h26.ts

question to the developper : why not put in a variable the same word in the different categories
for example 
UNKNOWN will always be UNKNOWN in all fields to be translated.
title, authors, price, copy, language ... well the list is long and the work of the translator could be lightened.

fewer errors because the same word will be the same throughout the interface.

technically it's not possible : too complex ?

**Comment by textbrowser** [comment]
Are you suggesting that I replace all of the tr("UNKNOWN") instances with a single instance?

**Reply by meteos77** [reply]
I am not a programmer : user example I set my program to locale=FR

the translators do once 
unknow_GB_gb="UNKNOW"
unknow_FR_fr="UNKNOW"
unknow_DE_de="UBEKANNT"

and that in your program you make unknow_$locale each time you want to display UNKNOW
is it possible in concept ?

**Comment by textbrowser** [comment]
If I have an unknown representation of something, I state it as such. :) Since it's translatable, it becomes tr("UNKNOWN"). I do not have instances of unknow_$locale. What's that?

https://doc.qt.io/qt-5/qtlinguist-hellotr-example.html

**Comment by textbrowser** [comment]
Alright, I am working on non-mandatory images. There is much work there. The translations are complete unless you find text which should be part of the translation process. Qt's system is clear: text that should be translated must be encased by tr(). I understand that there are many (several, more than one) instances of tr("UNKNOWN") and perhaps these should be replaced by a variable. I can address that separately.

**Comment by textbrowser** [comment]
:)

https://github.com/textbrowser/biblioteq/commit/c43ad4c2a1460597a502be53f7280ae94b12963b

**Comment by meteos77** [comment]
bravo for the support of so many machines/systems

I will test on raspberry when I have a moment to see the speed

for me your morning modifications were sufficient so I don't see too much progress made by the update tonight but I don't doubt that they are under the hood.

**Comment by textbrowser** [comment]
You must have a recent copy of the repository and compile a new build. It's slow because of life and there are significant changes with respect to the images and the progress dialogs. :P

**Comment by textbrowser** [comment]
> qmake -o Makefile biblioteq.arm.pro
> make -j $(nproc)

Or, you can download the Pi 4 package.

https://github.com/textbrowser/biblioteq/releases/download/2021.10.30/BiblioteQ-2021.10.30_armhf.deb

**Comment by textbrowser** [comment]
I'll let you know when you should test the image-less and progress-less changes. Also remember to disable images.

![Screenshot_20211108_130147](https://user-images.githubusercontent.com/10701156/140794165-bfbbf97c-0af8-4755-840d-82e2e7276024.png)

**Comment by meteos77** [comment]
I tried the package but I got an error related to a Qt library; it's probably my configuration

**Comment by textbrowser** [comment]
Let me try it.

**Comment by textbrowser** [comment]
pi@raspberrypi:~ $ lsb_release -a
No LSB modules are available.
Distributor ID: Raspbian
Description:    Raspbian GNU/Linux 11 (bullseye)
Release:        11
Codename:       bullseye
pi@raspberrypi:~ $ sudo apt install ./BiblioteQ-2021.10.30_armhf.deb 
[sudo] password for pi: 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'biblioteq' instead of './BiblioteQ-2021.10.30_armhf.deb'
The following additional packages will be installed:
  libopenjp2-7 libpoppler-qt5-1 libpoppler102 libxslt1.1 libyaz5 poppler-data
Suggested packages:
  poppler-utils ghostscript fonts-japanese-mincho | fonts-ipafont-mincho
  fonts-japanese-gothic | fonts-ipafont-gothic fonts-arphic-ukai fonts-arphic-uming
  fonts-nanum
The following NEW packages will be installed:
  biblioteq libopenjp2-7 libpoppler-qt5-1 libpoppler102 libxslt1.1 libyaz5 poppler-data
0 upgraded, 7 newly installed, 0 to remove and 0 not upgraded.
Need to get 4,014 kB/14.5 MB of archives.
After this operation, 20.0 MB of additional disk space will be used.
Do you want to continue? [Y/n] 
Get:1 /home/pi/BiblioteQ-2021.10.30_armhf.deb biblioteq armhf 2021.10.30 [10.5 MB]
Get:2 http://mirrors.ocf.berkeley.edu/raspbian/raspbian bullseye/main armhf poppler-data all 0.4.10-1 [1,602 kB]
Get:3 http://mirrors.ocf.berkeley.edu/raspbian/raspbian bullseye/main armhf libopenjp2-7 armhf 2.4.0-3 [150 kB]
Get:4 http://mirrors.ocf.berkeley.edu/raspbian/raspbian bullseye/main armhf libpoppler102 armhf 20.09.0-3.1 [1,521 kB]
Get:5 http://mirrors.ocf.berkeley.edu/raspbian/raspbian bullseye/main armhf libpoppler-qt5-1 armhf 20.09.0-3.1 [148 kB]
Get:6 http://mirrors.ocf.berkeley.edu/raspbian/raspbian bullseye/main armhf libxslt1.1 armhf 1.1.34-4 [218 kB]
Get:7 http://mirrors.ocf.berkeley.edu/raspbian/raspbian bullseye/main armhf libyaz5 armhf 5.30.3-1 [375 kB]
Fetched 4,014 kB in 7s (601 kB/s)                                                       
Selecting previously unselected package poppler-data.
(Reading database ... 65628 files and directories currently installed.)
Preparing to unpack .../0-poppler-data_0.4.10-1_all.deb ...
Unpacking poppler-data (0.4.10-1) ...
Selecting previously unselected package libopenjp2-7:armhf.
Preparing to unpack .../1-libopenjp2-7_2.4.0-3_armhf.deb ...
Unpacking libopenjp2-7:armhf (2.4.0-3) ...
Selecting previously unselected package libpoppler102:armhf.
Preparing to unpack .../2-libpoppler102_20.09.0-3.1_armhf.deb ...
Unpacking libpoppler102:armhf (20.09.0-3.1) ...
Selecting previously unselected package libpoppler-qt5-1:armhf.
Preparing to unpack .../3-libpoppler-qt5-1_20.09.0-3.1_armhf.deb ...
Unpacking libpoppler-qt5-1:armhf (20.09.0-3.1) ...
Selecting previously unselected package libxslt1.1:armhf.
Preparing to unpack .../4-libxslt1.1_1.1.34-4_armhf.deb ...
Unpacking libxslt1.1:armhf (1.1.34-4) ...
Selecting previously unselected package libyaz5:armhf.
Preparing to unpack .../5-libyaz5_5.30.3-1_armhf.deb ...
Unpacking libyaz5:armhf (5.30.3-1) ...
Selecting previously unselected package biblioteq.
Preparing to unpack .../6-BiblioteQ-2021.10.30_armhf.deb ...
Unpacking biblioteq (2021.10.30) ...
Setting up poppler-data (0.4.10-1) ...
Setting up libxslt1.1:armhf (1.1.34-4) ...
Setting up libopenjp2-7:armhf (2.4.0-3) ...
Setting up libyaz5:armhf (5.30.3-1) ...
Setting up libpoppler102:armhf (20.09.0-3.1) ...
Setting up libpoppler-qt5-1:armhf (20.09.0-3.1) ...
Setting up biblioteq (2021.10.30) ...
Processing triggers for fontconfig (2.13.1-4.2) ...
Processing triggers for libc-bin (2.31-13+rpi1+deb11u2) ...
Processing triggers for mailcap (3.69) ...
pi@raspberrypi:~ $ 

![Screenshot_20211108_130437](https://user-images.githubusercontent.com/10701156/140794559-9927b0c1-3258-4b64-8757-f51bf865bb2d.png)

**Comment by textbrowser** [comment]
![Screenshot_20211108_130641](https://user-images.githubusercontent.com/10701156/140794783-c382a342-d4d9-4a3e-97db-8a2a3496e55c.png)

**Comment by textbrowser** [comment]
How did you try to install it?

**Comment by meteos77** [comment]
I am debian 10
installation with your .deb
./BiblioteQ: relocation error: ./BiblioteQ: symbol _ZN9QSslError16staticMetaObjectE version Qt_5 not defined in file libQt5Network.so.5 with link time reference

**Comment by textbrowser** [comment]
The official release of Debian is 11. :)

**Comment by textbrowser** [comment]
Debian 10 includes Qt 5.11.3. The BQ in the package was built with Qt 5.12.8. Yah. I don't know the compatibility problems there. And I don't have the luxury to create packages for 10 and 11 or 9. If the package does not work, build from source.

**Comment by meteos77** [comment]
The official Debian version is 11 :-) for me it was released today and your package was done before.

https://www.phoronix.com/scan.php?page=news_item&px=Raspberry-Pi-OS-Bullseye

https://www.phoronix.com/scan.php?page=news_item&px=Raspberry-Pi-OS-Bullseyefile:

file:///home/yo/Documents/Debian11.png

**Reply by textbrowser** [reply]
The package was built on Raspbian 11 (Bullseye). You cannot install it on your system?

**Comment by meteos77** [comment]
https://www.raspberrypi.com/news/
je ne sais pas insérer une image dans github ?

**Comment by textbrowser** [comment]
So you're not using 10? So Debian 11 is at Qt 5.15.2. My build is from Qt 5.12.8. Anyway, there is a disconnect somewhere. Source then.

**Comment by meteos77** [comment]
I can't insert an image in github

**Comment by textbrowser** [comment]
See "Attach file by dragging & ..., selecting or pasting them." at the bottom of the input field.

**Comment by meteos77** [comment]
image test
no worries, I'll try to compile for the 10 in cross compiling ? this will teach me something new
then I'll move to Debian 11

**Comment by meteos77** [comment]
I retry an image
on the other hand dpkg did not flinch because the depandance is not up to date

I wish you a good end of day.

**Comment by meteos77** [comment]
retry an image
![This is an image]file:///home/yo/Documents/Debian11.png

**Reply by textbrowser** [reply]
You also. :)

The image is not visible.

**Comment by textbrowser** [comment]
UNKNOWN completed!

https://github.com/textbrowser/biblioteq/commit/009480b54fddbd09850c48e59ce0b94559891371

**Comment by meteos77** [comment]
wonderful

![image test]file:///home/yo/Documents/UNKNOWN-merveilleux.png

this method allows you to start processing 
d) translation is difficult,
from our topic: suggestion about internationalization / translation #117 
reducing from 18 to 1 the number of fields to translate is cool and allows a less tedious work for the translator.
remain the other fields : does it impact you a lot on the way you write your code ?

**Comment by textbrowser** [comment]
Which other fields? In UI files, that is not recommended as those files are governed by Qt's Designer. Do you have a list of other duplicates?

**Reply by meteos77** [reply]
I looked more precisely even if I understood that it is not advised

ex : 
13 Database Error
20 Title
13 BiblioteQ: Progress Dialog
19 Price
11 Music C
... 
[liste-traduction-repetitive.txt](https://github.com/textbrowser/biblioteq/files/7506922/liste-traduction-repetitive.txt)

cat liste-traduction-repetitive.txt | wc -l
2220

cat liste-traduction-repetitive.txt | uniq | wc -l
1294

difference 926 translations for the translator !

**Comment by textbrowser** [comment]
See this image.

![Screenshot_20211109_133430](https://user-images.githubusercontent.com/10701156/140984000-6adbbd06-9418-4e0b-8402-0272ce6f8175.png)

I do not have the interest to replace every duplicate translatable text in the source with common containers. Linguist provides a replace-all mechanism.

**Comment by textbrowser** [comment]
Alright, you desire a new import. The translation process is a responsibility of the translator. I have replaced one duplicate item (UNKNOWN). It would not have been a difficult task to use the Linguist to replace all of the instances. It required more work from me. :) So, let's focus on important matters. Complete the important requests and focus on simplifying things at other times.

**Comment by meteos77** [comment]
addition to my to-do list: read the documentation of qt linguist to be more efficient

you are absolutely right to remind me of the priorities so
sorry to put scattered, it's that I'm too entousist :-)

you are absolutely right to remind me of the priorities so
sorry for spreading myself too thin, I'm just too enthusiastic :-)

priority number 1 is the import of unimarc files with fields 995. it's a big deal...
